dcos_test_utils/gce.py

```diff
@@ -0,0 +1,254 @@
+            'name': deployment_name,
+            'target': {
+                'config': {
+                    'content': resources
+                }
+            }
+        }
+
+        response = self.deployment_manager.deployments().insert(project=self.project_id, body=body).execute()
+        log.debug('GceWrapper: create_deployment response: ' + str(response))
+
+    @catch_http_exceptions
+    def get_resources_info(self, name):
+        request = self.deployment_manager.resources().list(project=self.project_id, deployment=name)
+        while request is not None:
+            response = request.execute()
+            log.debug('GceWrapper: get_resources_info response: ' + str(response))
+
+            for resource_info in response['resources']:
+                yield resource_info
+
+            request = self.deployment_manager.resources().list_next(previous_request=request,
+                                                                    previous_response=response)
+
+    @catch_http_exceptions
+    def get_instance_info(self, name, zone):
+        response = self.compute.instances().get(project=self.project_id, zone=zone, instance=name).execute()
+        log.debug('GceWrapper: get_instance_info response: ' + str(response))
+        return response
+
+    @catch_http_exceptions
+    def get_deployment_info(self, name) -> dict:
+        """ Returns the dictionary representation of a GCE deployment resource. For details on the contents of this
+            resource, see https://cloud.google.com/deployment-manager/docs/reference/latest/deployments#resource"""
+        response = self.deployment_manager.deployments().get(project=self.project_id,
+                                                             deployment=name).execute()
+        log.debug('GceWrapper: get_deployment_info response: ' + str(response))
+        return response
+
+    @catch_http_exceptions
+    def delete_deployment(self, name):
+        response = self.deployment_manager.deployments().delete(project=self.project_id, deployment=name).execute()
+        log.debug('GceWrapper: delete_deployment response: ' + str(response))
+
+    @catch_http_exceptions
+    def list_group_instances(self, group_name, zone) -> typing.Iterator(dict):
+        response = self.compute.instanceGroupManagers().listManagedInstances(project=self.project_id, zone=zone,
+                                                                             instanceGroupManager=group_name).execute()
+        log.debug('GceWrapper: list_group_instances response: ' + str(response))
+
+        for instance in response['managedInstances']:
+            yield instance
+
+
+class Deployment:
+    def __init__(self, gce_wrapper, name, zone):
+        self.gce_wrapper = gce_wrapper
+        self.name = name
+        self.zone = zone
+        self.instance_group_name = self.name + '-group'
+
+    def create(self, num_masters, num_public_agents, num_private_agents, source_image, machine_type, image_project,
```

Comment:
Indeed, although I need instance specific attributes in that method such as the deployment's name and zone, so perhaps we should eliminate the create method entirely and just put the code in the Deployment's __init__?